File: practice3/tcastellanosa/backend/src/Application/DTOs/ProductDtos.cs
using System.ComponentModel.DataAnnotations;

namespace Application.DTOs;

public record ProductDto(Guid Id, string Name, string? Description, decimal Price, int Stock, bool IsActive, DateTime CreatedAt, Guid? CategoryId, string? CategoryName);
public record CreateProductRequest([param: Required, StringLength(150)] string Name, [param: StringLength(1000)] string? Description, [param: Range(typeof(decimal), "0.01", "999999999")] decimal Price, [param: Range(0, int.MaxValue)] int Stock, bool IsActive, Guid? CategoryId = null);
public record UpdateProductRequest([param: Required, StringLength(150)] string Name, [param: StringLength(1000)] string? Description, [param: Range(typeof(decimal), "0.01", "999999999")] decimal Price, [param: Range(0, int.MaxValue)] int Stock, bool IsActive, Guid? CategoryId = null);

public record ProductPagedResult(IEnumerable<ProductDto> Items, int TotalItems, int Page, int PageSize, int TotalPages);

File: Labo 2/catalog/src/main/java/com/enterprise/catalog/dto/ProductDtos.java
package com.enterprise.catalog.dto;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;

import java.math.BigDecimal;
import java.time.Instant;

public class ProductDtos {

    public record ProductResponse(
            String id,
            String name,
            String description,
            BigDecimal price,
            int stock,
            boolean isActive,
            String categoryId,
            String categoryName,
            Instant createdAt
    ) {
    }

    public record ProductRequest(
            @NotBlank(message = "Name is required") String name,
            String description,
            @DecimalMin(value = "0.0", inclusive = true, message = "Price must be >= 0") BigDecimal price,
            @Min(value = 0, message = "Stock must be >= 0") int stock,
            boolean isActive,
            String categoryId
    ) {
    }

    public record PagedResponse<T>(
            java.util.List<T> items,
            long totalItems,
            int page,
            int pageSize,
            int totalPages
    ) {
    }
}
